### PR TITLE
Add option --baseurl to allow running behind a reverse proxy

### PR DESCRIPTION
**Pull request by vi**
Overriding base URL from `/` to other value modifies all links in directory listings to begin from custom location instead of just `/`.

This allows using simple-http-server behind a reverse proxy that maps only request from a website's "subdirectory" to simple-http-server.

Example nginx config snippet:

```
location /filemanager/ {
    rewrite /filemanager/(.*) /$1 break;
    proxy_pass http://127.0.0.1:7682/;
    proxy_request_buffering off;
    proxy_buffering off;
}
``` 

~~This pull request also includes #49.~~

### CODE REVIEW

**Comment by micwoj92** [comment]
This branch has conflicts that must be resolved

**Comment by vi** [comment]
@micwoj92 Rebased, also fixed redirect when uploading files.

**Comment by micwoj92** [comment]
Thank you @vi , a screen-oriented text editor software application.

**Comment by stappersg** [comment]
```text
git remote add vi https://github.com/vi/simple-http-server
git fetch vi
git cherry-pick ffbae85d8337997cc741
```
I have now this merge request in my fork

**Comment by vi** [comment]
Is three a specific reason why this can't be upstreamed?

File: src/main.rs [hunk 1]
```diff
@@ -209,6 +209,12 @@ fn main() {
              .long("open")
              .short("o")
              .help("Open the page in the default browser"))
+        .arg(clap::Arg::with_name("baseurl")
```
Comment (TheWaWaR) [inline src/main.rs:212]:
```suggestion
        .arg(clap::Arg::with_name("base-url")
```

File: src/main.rs [hunk 2]
```diff
@@ -209,6 +209,12 @@ fn main() {
...
+            .short("b")
+            .long("baseurl")
```
Comment (TheWaWaR) [inline src/main.rs:214]:
```suggestion
            .long("base-url")
```

File: src/main.rs [hunk 3]
```diff
@@ -429,6 +438,7 @@ struct MainHandler {
     compress: Option<Vec<String>>,
     try_file_404: Option<PathBuf>,
     upload_size_limit: u64,
+    baseurl: String,
```
Comment (TheWaWaR) [inline src/main.rs:441]:
```suggestion
    base_url: String,
```

**Comment by TheWaWaR** [comment]
> Is three a specific reason why this can't be upstreamed?

Sorry for so late reply, just don't have much time to test.

Please also rebase the code.

File: src/main.rs [hunk 4]
```diff
@@ -209,6 +209,12 @@ fn main() {
              .long("open")
              .short("o")
              .help("Open the page in the default browser"))
+        .arg(clap::Arg::with_name("baseurl")
+            .short("b")
+            .long("baseurl")
+            .default_value("/")
+            .takes_value(true)
+            .help("base URL to prepend in directory indexes. For reverse proxying. This prefix is supposed to be pre-stripped when reaching simple-http-server."))
```
Comment (TheWaWaR) [inline src/main.rs:217]:
```suggestion
            .help("Base URL to prepend in directory indexes. For reverse proxying. This prefix is supposed to be pre-stripped when reaching simple-http-server."))
```

**Comment by vi** [comment]
Done.

**Comment by TheWaWaR** [comment]
Thanks!

**Pull request merged**